io.openems.edge.goodwe.et/src/io/openems/edge/goodwe/et/ess/GoodWeEtBatteryInverterImpl.java

```diff
@@ -393,34 +397,26 @@ protected ModbusProtocol defineModbusProtocol() {
 	public void applyPower(int activePower, int reactivePower) throws OpenemsNamedException {
 		final PowerModeEms nextPowerMode;
 
+		System.out.println("Active power:" + activePower);
+
 		if (this.config.readOnlyMode()) {
 			// Read-Only-Mode: fall-back to internal self-consumption optimization
 			nextPowerMode = PowerModeEms.AUTO;
 			activePower = 0;
 		} else {
-			if (activePower <= 0) {
-				// ActivePower is negative or zero -> CHARGE
+			if (activePower < 0) {
+				// ActivePower is negative -> CHARGE
+				System.out.println(" Setting Charging");
 				nextPowerMode = PowerModeEms.CHARGE_BAT;
 
 			} else {
-				// ActivePower is positive -> DISCHARGE
-
-				/*
-				 * Check if PV is available. Discharge mode changes according to availability of
-				 * PV
-				 * 
-				 * TODO PV mode is not working, need an update from GoodWe for this.
-				 */
-				Integer productionPower = null;
-				for (AbstractGoodWeEtCharger charger : this.chargers) {
-					productionPower = TypeUtils.sum(productionPower, charger.getActualPower().get());
-				}
-				if (productionPower == null) {
-					// No PV-Power -> required to put on SELL_POWER
-					nextPowerMode = PowerModeEms.SELL_POWER;
+				// ActivePower is positive or zero -> DISCHARGE
+				System.out.println(" Setting Discharging");
+				Integer soc = this.getSoc().get();
```

Comment:
@venu-sagar: this obivously still throws a NullPointerException if Soc is UNDEFINED (i.e. at start or when modbus connection is broken)